gradle-repository/buildSrc/src/main/kotlin/gradlebuild.gradle-api-repository.gradle.kts

```diff
@@ -0,0 +1,85 @@
+plugins {
+    base
+    `maven-publish`
+}
+
+val gradleApiRepository = extensions.create<GradleRepositoryExtension>("gradleApiRepository")
+
+gradleApiRepository.apply {
+    gradleVersion.convention(
+        providers.gradleProperty("gradleApiVersion").orElse("7.4.1")
+    )
+    repoLocation.convention(
+        layout.buildDirectory.dir("repo")
+    )
+}
+
+repositories {
+    listOf("distributions", "distributions-snapshots").forEach { distUrl ->
+        ivy {
+            name = "Gradle distributions"
+            url = uri("https://services.gradle.org")
+            patternLayout {
+                artifact("/${distUrl}/[module]-[revision]-bin(.[ext])")
+            }
```

Comment:
Does using an Ivy layout have some benefits here? If not, wouldn't a Maven one be simpler?

Reply:
Ivy layout in this case is a workaround for "Use an arbitrary URL and cache the result". You can't use a Maven layout, since a Maven layout can't be customized.

Reply:
Why do we need to customize it?